src/metabase/cmd/rotate_encryption_key.clj

```diff
@@ -19,42 +18,32 @@
     (log/warnf "Database not found. Metabase will create a new database at %s and proceeed encrypting." "2")
     (mdb/setup-db!))
   (log/infof "%s: %s | %s" (trs "Connected to") mdb.env/db-type (mdb.env/db-file))
-  (let [make-encrypt-fn   (fn [maybe-encrypt-fn]
-                            (if to-key
-                              (partial maybe-encrypt-fn (encryption/validate-and-hash-secret-key to-key))
-                              identity))
-        encrypt-str-fn    (make-encrypt-fn encryption/maybe-encrypt)
-        encrypt-bytes-fn  (make-encrypt-fn encryption/maybe-encrypt-bytes)
-        is-h2             (= :h2 (mdb/db-type))
-        value-column      (if is-h2
-                            "\"VALUE\""
-                            :value)
-        setting-where     (if is-h2
-                            "setting.\"KEY\" = ?"
-                            "setting.key = ?")]
+  (let [make-encrypt-fn  (fn [maybe-encrypt-fn]
+                           (if to-key
+                             (partial maybe-encrypt-fn (encryption/validate-and-hash-secret-key to-key))
+                             identity))
+        encrypt-str-fn   (make-encrypt-fn encryption/maybe-encrypt)
+        encrypt-bytes-fn (make-encrypt-fn encryption/maybe-encrypt-bytes)]
     (t2/with-transaction [t-conn {:datasource (mdb.connection/data-source)}]
       (doseq [[id details] (t2/select-pk->fn :details Database)]
         (when (encryption/possibly-encrypted-string? details)
           (throw (ex-info (trs "Can''t decrypt app db with MB_ENCRYPTION_SECRET_KEY") {:database-id id})))
-        (jdbc/update! {:connection t-conn}
-                      :metabase_database
-                      {:details (encrypt-str-fn (json/encode details))}
-                      ["metabase_database.id = ?" id]))
+        (t2/update! :conn t-conn :metabase_database
```

Comment:
the rotate key test was failing ([job](https://github.com/metabase/metabase/actions/runs/6116817750/job/16602552829#step:7:198)).

I'm not sure why it can't recognize the save point when using `jdbc`, maybe `jdbc/update! implicit commit the transaction?

IDK, but we already use the t2 API here so I don't know why we need to use the low level JDBC API.